### PR TITLE
polylith: 0.1.0-alpha9 -> 0.2.12-alpha

### PR DESCRIPTION
**Pull request by ericdallo**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 139463` at 2339ab84 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>polylith</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `nix-update-script`.
Near pkgs/development/tools/misc/polylith/default.nix:5:3:
```
  |
5 | , nix-update-script
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/misc/polylith/default.nix [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 
 stdenv.mkDerivation rec {
   pname = "polylith";
-  version = "0.1.0-alpha9";
+  version = "v0.2.12-alpha";
 
   src = fetchurl {
-    url = "https://github.com/polyfy/polylith/releases/download/v${version}/poly-${version}.jar";
-    sha256 = "0mjn0fibj7z8wihk5frhyd5ai2bmzm909701sphjs7j9lgg0gc4k";
+    url = "https://github.com/polyfy/polylith/releases/download/${version}/poly-${lib.removePrefix "v" version}.jar";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/polylith/default.nix:14]:
```suggestion
    url = "https://github.com/polyfy/polylith/releases/download/v${version}/poly-$version}.jar";
```

File: pkgs/development/tools/misc/polylith/default.nix [hunk 2]
```diff
@@ -34,11 +40,22 @@ stdenv.mkDerivation rec {
   installCheckPhase = ''
     runHook preInstallCheck
 
-    $out/bin/poly help | fgrep -q '${version}'
+    $out/bin/poly help | fgrep -q '${lib.removePrefix "v" version}'
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/polylith/default.nix:43]:
```suggestion
    $out/bin/poly help | fgrep -q '${version}'
```

File: pkgs/development/tools/misc/polylith/default.nix [hunk 3]
```diff
@@ -34,11 +40,22 @@ stdenv.mkDerivation rec {
...
+  passthru.updateScript = writeScript "update-clojure-lsp" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i bash -p curl common-updater-scripts gnused jq nix
+
+    set -eu -o pipefail
+
+    latest_version=$(curl -s https://api.github.com/repos/polyfy/polylith/releases/latest | jq --raw-output .tag_name)
+
+    update-source-version polylith "$latest_version"
+  '';
```
[the remark this answers is not in the document]
Comment (ericdallo) [reply pkgs/development/tools/misc/polylith/default.nix:57]: I didn't know about that one, thanks, will try

**Comment by ericdallo** [comment]
Closing as this was already made on https://github.com/NixOS/nixpkgs/commit/063fde88482a8c78205e16f8dc4d29b3f98857dd

**Pull request closed without merging**